pkg/nsx/services/firewall.go

```diff
@@ -260,17 +327,50 @@ func (service *SecurityPolicyService) buildConjOperator(op string) *data.StructV
 	return operator
 }
 
-func (service *SecurityPolicyService) buildExpression(resource_type, member_type, value, key, operator string) *data.StructValue {
-	expression := data.NewStructValue(
-		"",
-		map[string]data.DataValue{
-			"resource_type": data.NewStringValue(resource_type),
-			"member_type":   data.NewStringValue(member_type),
-			"value":         data.NewStringValue(value),
-			"key":           data.NewStringValue(key),
-			"operator":      data.NewStringValue(operator),
-		},
-	)
+func (service *SecurityPolicyService) buildGroupExpression(policyExpression *[]*data.StructValue) *data.ListValue {
+	if policyExpression != nil {
+		expressions := data.NewListValue()
+		expressionFrame := data.NewStructValue(
+			"",
+			map[string]data.DataValue{
+				"expressions":   expressions,
+				"resource_type": data.NewStringValue("NestedExpression"),
+			},
+		)
+		*policyExpression = append(*policyExpression, expressionFrame)
+		return expressions
+	} else {
+		return nil
```

Comment:
This cannot return nil(although I'm not sure if the case of returning nil will happen) because the returned var will invoke `Add` method. At least it should be an empty list?

Reply:
Agreed, Thanks, corrected.